core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,319 @@
+    var isPresent = await blogDashboardIntroMessage.isPresent();
+    if (!isPresent) {
+      await this.waitForDraftsBlogPostsToLoad();
+    }
+    expect(await blogPostListItems.count()).toBe(number);
+  };
+
+  this.navigateToBlogPostEditorWithIndexFromList = async function(index) {
+    await action.click('blog post row', blogPostListItems.get(index));
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.getBlogPostCount = async function() {
+    return await BlogPostListItems.count();
+  };
+
+  this.deleteBlogPostWithIndex = async function(index) {
+    await action.click(
+      'Blog post edit option', BlogPostEditOptions.get(index));
+    await action.click('Delete blog post button', deleteBlogPostButton);
+    await action.click(
+      'Confirm Delete Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast('Blog Post Deleted Successfully.');
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.deleteBlogPostFromEditor = async function() {
+    await waitFor.visibilityOf(
+      deleteBlogPostButton, 'Unable to find delete button');
+    await action.click('Delete blog post button', deleteBlogPostButton);
+    await action.click(
+      'Confirm Delete Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast('Blog Post Deleted Successfully.');
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.unpublishBlogPostWithIndex = async function(index) {
+    await this.waitForPublishedBlogPostsToLoad();
+    await action.click(
+      'Blog post edit option', BlogPostEditOptions.get(index));
+    await action.click('Unpublish blog post button', unpublishBlogPostButton);
+    await action.click(
+      'Confirm unpublishing Blog Post button', confirmButton);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.navigateToBlogPostEditorWithIndex = async function(index) {
+    await action.click(
+      'Blog post edit option', BlogPostEditOptions.get(index));
+    await action.click(
+      'Edit blog post button', editBlogPostButton);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.getListView = async function() {
+    await waitFor.visibilityOf(
+      listViewButton, 'Unable to find list view button');
+    await action.click('List View Button', listViewButton);
+  };
+
+  this.getTilesView = async function() {
+    await waitFor.visibilityOf(
+      tilesViewButton, 'Unable to find tiles view button');
+    await action.click('Tiles View Button', tilesViewButton);
+  };
+
+  this.expectCurrUserToHaveProfilePhoto = async function() {
+    await waitFor.visibilityOf(
+      currUserProfilePhoto,
+      'Current user profile photo taking too long to display');
+  };
+
+  this.expectCurrUsernameToBeVisible = async function() {
+    await waitFor.visibilityOf(
+      currUsername,
+      'Current user name taking too long to display');
+  };
+
+  this.expectUploadError = async function() {
+    expect(await thumbnailUploadError.isDisplayed()).toBe(true);
```

Comment:
Can you use a waitFor here instead in case it takes some time to load?

Reply:
Done.